src/main/common/maths.c

```diff
@@ -210,14 +210,28 @@ void buildRotationMatrix(fp_angles_t *delta, fp_rotationMatrix_t *rotation)
     rotation->m[2][Z] = cosy * cosx;
 }
 
-void applyMatrixRotation(float *v, fp_rotationMatrix_t *rotationMatrix)
+void transposeMatrix3x3(const float matrix[3][3], float transposedMatrix[3][3])
+{
+    for (int i = 0; i < 3; i++) {
+        for (int j = 0; j < 3; j++) {
+            transposedMatrix[i][j] = matrix[j][i];
+        }
+    }
+}
+
+void applyMatrixRotationArray(float *v, float rotationMatrix[3][3])
```

Comment:
This is not necessary. Also there is PR for vector math refactoring, it may be better to wait until it is merged